File: src/main/java/aurocosh/divinefavor/common/damage_source/FrostDamageSource.java
package aurocosh.divinefavor.common.damage_source;

import net.minecraft.entity.EntityLivingBase;
import net.minecraft.util.DamageSource;
import net.minecraft.util.text.ITextComponent;
import net.minecraft.util.text.TextComponentTranslation;

public class FrostDamageSource extends DamageSource {
    public FrostDamageSource() {
        super("frost");
        setDamageBypassesArmor();
        setMagicDamage();
    }

    public ITextComponent getDeathMessage(EntityLivingBase entityLivingBaseIn) {
        String s = "death.attack." + this.damageType;
        return new TextComponentTranslation(s, entityLivingBaseIn.getDisplayName());
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/damage_source/FrostDamageSource.kt
package aurocosh.divinefavor.common.damage_source

import net.minecraft.entity.EntityLivingBase
import net.minecraft.util.DamageSource
import net.minecraft.util.text.ITextComponent
import net.minecraft.util.text.TextComponentTranslation

class FrostDamageSource : DamageSource("frost") {
    init {
        setDamageBypassesArmor()
        setMagicDamage()
    }

    override fun getDeathMessage(entityLivingBaseIn: EntityLivingBase): ITextComponent {
        val s = "death.attack." + this.damageType
        return TextComponentTranslation(s, entityLivingBaseIn.displayName)
    }
}